docs/testplan/platform/pmon-test-plan.md

```diff
@@ -0,0 +1,123 @@
+* Perform the following steps for the list of the expected daemon
+	* Verify the daemon is running
+	* Restart the daemon
+	* Verify the daemon is running again
+
+### Test case # 3 – Data for chassisd daemon
+#### Test objective
+Verify the expected data for chassisd daemon (if not skip_chassisd)
+#### Test steps
+* Verify the chassisd running status based on the platform type (if IS_MODULAR_CHASSIS == 1 : "/usr/share/sonic/platform/chassisdb.conf" )
+	* Verify that chassisd is not running for the fixed platform otherwise go next
+	* Verify that module db data setting
+```
+CHASSIS_INFO_TABLE = 'CHASSIS_TABLE'
+CHASSIS_INFO_KEY_TEMPLATE = 'CHASSIS {}'
+CHASSIS_INFO_CARD_NUM_FIELD = 'module_num'
+
+CHASSIS_MODULE_INFO_TABLE = 'CHASSIS_MODULE_TABLE'
+CHASSIS_MODULE_INFO_KEY_TEMPLATE = 'CHASSIS_MODULE {}'
+CHASSIS_MODULE_INFO_NAME_FIELD = 'name'
+CHASSIS_MODULE_INFO_DESC_FIELD = 'desc'
+CHASSIS_MODULE_INFO_SLOT_FIELD = 'slot'
+CHASSIS_MODULE_INFO_OPERSTATUS_FIELD = 'oper_status'
+```
+
+### Test case # 4 – Data for ledd daemon
+#### Test objective
+Verify the expected data for ledd daemon (if not skip_ledd)
+#### Test steps
+* Verify the ledd running status
+
+### Test case # 5 – Data for pcied daemon
+#### Test objective
+Verify the expected data for pcied daemon (if not skip_pcied)
+#### Test steps
+* Verify the pcie config file "/usr/share/sonic/platform/pcie.yaml" exists
+* Verify the "PCIE_DEVICES" has "status" field set to "SUCCESS" 
+
+### Test case # 6 – Data for psud daemon
+#### Test objective
+Verify the expected data for psud daemon (if not skip_psud)
+#### Test steps
+* Verify the 'PSU_INFO|PSU {}' is/are available
+* Verify the information data for each 'PSU {}' is expected to compare "/usr/share/sonic/platform/platform.json"
+
+### Test case # 7 – Data for syseepromd daemon
+#### Test objective
+Verify the expected data for syseepromd daemon (if not skip_syseepromd)
+#### Test steps
+* Verify the 'EEPROM_INFO|State' in state_db has the 'Initialized' field set to '1'
+* Verify the 'EEPROM_INFO|*' has valid pairs of key and value
+   * Q) how to verify the data is valid value or if we can skip the check for the value?
+
+### Test case # 8 – Data for thermalctld daemon
+#### Test objective
+Verify the expected data for thermalctld daemon (if not skip_thermalctld)
+#### Test steps
+* Verify the policy file "/usr/share/sonic/platform/thermal_policy.json" exists
+* Verify the 'FAN_INFO|{}' and 'TEMPERATURE_INFO|{}' available
+   * Q) how to verify the data is valid value ? log check?
+
+### Test case # 9 – Data for xcvrd daemon
+#### Test objective
+Verify the expected data for xcvrd daemon (if not skip_xcvrd)
+#### Test steps
+* Verify the 'TRANSCEIVER_INFO', 'TRANSCEIVER_DOM_SENSOR' and 'TRANSCEIVER_STATUS' state_db table has data
+* Verify the information data for each 'TRANSCEIVER_INFO' is expected to compare "/usr/share/sonic/platform/platform.json"
+   * Q) how to verify the data is valid value? log check?
```

Comment:
We can call the platform API directly and compare values. For sensor values, we would need to allow for some tolerance, as these values can change often.